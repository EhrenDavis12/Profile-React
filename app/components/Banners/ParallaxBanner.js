import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { Parallax } from 'react-parallax';
import Header from 'components/Banners/Header';
import Banner from 'components/Banners/Banner';

const Container = styled.section`
  margin-bottom: 40px;
`;

const PaddingSections = styled.div`
  margin-bottom: 20px;
  margin-top: 20px;
  text-align: center;
  font-size: 2em;
`;

function SectionBanner(props) {
  return (
    <React.Fragment>
      <Parallax
        bgImage={props.image}
        strength={props.strength || 500}
        // bgImageStyle={{ width: '90%' }} // hight width: '90%'
      >
        <Container className="container">
          <Banner className="row">
            <div id={props.id}>
              <Header>{props.header}</Header>
            </div>
          </Banner>
          {props.children}
        </Container>
      </Parallax>
      <PaddingSections>| | |</PaddingSections>
    </React.Fragment>
  );
}

SectionBanner.propTypes = {
  children: PropTypes.node.isRequired,
  image: PropTypes.node.isRequired,
  header: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  strength: PropTypes.number,
};

export default SectionBanner;
