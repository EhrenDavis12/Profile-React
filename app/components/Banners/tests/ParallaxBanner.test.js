import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import BkgImg6 from 'images/BkgBanners/small_bluePur_sky.jpg';
import ParallaxBanner from '../ParallaxBanner';

describe('<ParallaxBanner />', () => {
  it('should render the page default Strength', () => {
    const renderedComponent = shallow(
      <ParallaxBanner image={BkgImg6} id="Hobbies" header="Hobbies">
        <div>Hello World</div>
      </ParallaxBanner>,
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the page given Strength', () => {
    const renderedComponent = shallow(
      <ParallaxBanner
        image={BkgImg6}
        id="Hobbies"
        header="Hobbies"
        strength={1000}
      >
        <div>Hello World</div>
      </ParallaxBanner>,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
