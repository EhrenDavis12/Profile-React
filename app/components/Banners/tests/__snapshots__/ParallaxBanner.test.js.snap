// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParallaxBanner /> should render the page default Strength 1`] = `
<Fragment>
  <t
    bgClassName="react-parallax-bgimage"
    bgImage="IMAGE_MOCK"
    bgImageAlt=""
    className=""
    contentClassName=""
    disabled={false}
    log={false}
    strength={500}
  >
    <ParallaxBanner__Container
      className="container"
    >
      <BannerSection
        className="row"
      >
        <div
          id="Hobbies"
        >
          <Header>
            Hobbies
          </Header>
        </div>
      </BannerSection>
      <div>
        Hello World
      </div>
    </ParallaxBanner__Container>
  </t>
  <ParallaxBanner__PaddingSections>
    | | |
  </ParallaxBanner__PaddingSections>
</Fragment>
`;

exports[`<ParallaxBanner /> should render the page given Strength 1`] = `
<Fragment>
  <t
    bgClassName="react-parallax-bgimage"
    bgImage="IMAGE_MOCK"
    bgImageAlt=""
    className=""
    contentClassName=""
    disabled={false}
    log={false}
    strength={1000}
  >
    <ParallaxBanner__Container
      className="container"
    >
      <BannerSection
        className="row"
      >
        <div
          id="Hobbies"
        >
          <Header>
            Hobbies
          </Header>
        </div>
      </BannerSection>
      <div>
        Hello World
      </div>
    </ParallaxBanner__Container>
  </t>
  <ParallaxBanner__PaddingSections>
    | | |
  </ParallaxBanner__PaddingSections>
</Fragment>
`;
